### PR TITLE
Propose subvolumes (WIP)

### PR DESCRIPTION
**Pull request by shundhammer**
https://trello.com/c/Zboq9Byg/495-5-storageng-proposal-propose-btrfs-subvolumes

Screenshot from integration test with DUD from this package on the latest StorageNG ISO:

storage-ng-subvol-proposal

### CODE REVIEW

File: src/lib/y2storage/fake_device_factory.rb [hunk 1]
```diff
@@ -592,6 +597,64 @@ def create_lvm_pv(parent, args)
...
+    def create_btrfs(parent, args)
+      log.info("#{__method__}( #{parent}, #{args} )")
+      default_subvolume = args["default_subvolume"]
+      if default_subvolume
+        blk_device = ::Storage::BlkDevice.find_by_name(@devicegraph, parent)
+        filesystem = blk_device.filesystem
+        if !filesystem || filesystem.type != ::Storage::FsType_BTRFS
+          raise HierarchyError, "No btrfs on #{parent}"
+        end
+        filesystem.create_btrfs_subvolume(@default_subvolume)
```
Comment (ancorgs) [inline src/lib/y2storage/fake_device_factory.rb]: Chunk of code adapted to the wrapper
```ruby
blk_device = Y2Storage::BlkDevice.find_by_name(@devicegraph, parent)
filesystem = blk_device.filesystem
if !filesystem || !filesystem.type.is?(:btrfs)
  raise HierarchyError, "No btrfs on #{parent}"
end
filesystem.create_btrfs_subvolume(@default_subvolume)
```

Comment (shundhammer) [inline src/lib/y2storage/fake_device_factory.rb]: OK, thanks, fixed.

File: src/lib/y2storage/planned_volume.rb [hunk 2]
```diff
@@ -188,11 +188,12 @@ def create_filesystem(partition)
     #
     # @param filesystem [::Storage::Filesystem]
     def create_subvolumes(filesystem)
-      return unless filesystem.btrfs?
+      return unless filesystem.type == ::Storage::FsType_BTRFS
```
Comment (wfeldt) [inline src/lib/y2storage/planned_volume.rb]: shouldn't this look like "filesystem.type.is?(:btrfs)" now?

**Comment by wfeldt** [comment]
apart from the minor issue and the rubocop offense, lgtm

File: src/lib/y2storage/planned_volume.rb (hunk 2 above)
Comment (shundhammer) [inline src/lib/y2storage/planned_volume.rb]: Yes, but this doesn't use the new API at all yet. We'll get to that at some later time.

**Pull request merged**